Work Summaries

SondraE opened the pull request:
![boris-smokrovic-lyvCvA8sKGc-unsplash](https://user-images.githubusercontent.com/8933115/124824451-9e772f00-df2f-11eb-8118-815ed66157ab.jpg)


## Description
Most of our projects before 2020 have descriptions of the company we worked for and what the tool we built does, but it's not clear what services OddBird provided. This is an attempt to make that info clear and easy to read.


## Steps to test/reproduce
/work/shopify/
/work/cfoshare/
/work/workflow-builder/
/work/expression-builder/

SondraE commented:
@mirisuzanne @stacyk This is nowhere near finished. I'm curious to get your opinions on whether I'm going in the right direction with the layout of the content.

SondraE commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Rewrite summaries for client work (Oddleventy & Marketing PDF) to focus more on what OddBird did and less on what the client's application does.](https://trello.com/c/0iZ7AyVL/64-rewrite-summaries-for-client-work-oddleventy-marketing-pdf-to-focus-more-on-what-oddbird-did-and-less-on-what-the-clients-applic)

stacyk commented on content/work/cfoshare.md:
```diff
@@ -27,11 +27,23 @@ summary: |
-CFOshare is a "fractional CFO"
-that serves as an outsourced finance and accounting department
-for a range of businesses.
+**Dates**: 2018 - 2020 
+
+**Sector**: Finance & Accounting
+
+**What We Did**: 
+- Transitioned website to WordPress CMS
+- Brand implementation
+- Content migration
+- Image optimization
+- Trained staff in on-going maintenance  
+
```
What if we did something here like we did on the Design Services page (I plopped this content in this space, but we'd use another image)![Screen Shot 2021-07-08 at 3 52 09 PM](https://user-images.githubusercontent.com/1581694/124989789-224f1b00-e005-11eb-8c97-44b951ba9f14.jpg)


Orignial can be found here:
https://deploy-preview-167--oddleventy.netlify.app/services/development/

The original section looks like this in code if this is helpful:

```
{% import 'embed.macros.njk' as embed %}
{% import 'layout.macros.njk' as layout %}


{{ layout.title('Services') }}

{% call embed.media_block(
  media=embed.screen(
    notch='0',
    content=embed.img(
      src='pages/trig-phone.jpg',
      alt='trig race screen',
      width=744 ,
      height=1611,
      sizes='media'
    )
  )
) %}

{% call layout.grid('narrow-columns') %}

{% call layout.block('column') %}

### Design

- User Experience (UX) - OOUX Certified
- User Interface (UI) & Interaction
- Logos & Branding
- Integrated Design Systems
- Usability Testing
- Information Architecture

{% endcall %}

{% call layout.block('column') %}

### Development

- Full-stack Frontend & Backend
- 100% Test Coverage
- Full Documentation
- Automated Pattern Libraries
- WCAG Accessibility
- Refactors & Performance

{% endcall %}

{% endcall %}

{% endcall %}
```

stacyk reviewed (commented):
@SondraE Looking good. I like the addition of what we actually did. I left a formatting suggestion, but if that doesn't work, then maybe we could add the "What we did" as a heading instead of  bold text.  That might be a little odd though if dates and sectors are not headings, but those feel less substantial as a thing that would need a heading.

mirisuzanne commented:
I like the idea here - but I agree with Stacy that we can do better on the formatting.

The "dates" item in particular feels to me like a duplicate of the date we already have in the header. Could we just change the header to show a range, rather than a single date?

(I also wonder if the "sector" would make sense in the header somewhere, but I'm less sure of that)

SondraE commented:
@mirisuzanne @stacyk 
I've tried several redesign options here:

- Changed date to date range in the sub-header
- Added sector to the sub-header
- Added sector tags so that people can click on them to find all the clients in one sector
- Added "Work Sample" tag, but not sure how useful that one is. Leaning toward removing it.
- Added "What We Did" section below the summary.
- I think all the summaries should begin with (basically a tl;dr of) what we did, followed by a brief description of the client's software like I show in the CFOshare screen shot.
<img width="1440" alt="Screen Shot 2021-07-13 at 1 48 25 PM" src="https://user-images.githubusercontent.com/8933115/125516084-73d09a95-fa22-4924-99ad-2d7856bcf993.png">
<img width="788" alt="Screen Shot 2021-07-13 at 1 49 02 PM" src="https://user-images.githubusercontent.com/8933115/125516095-2158b4b9-f1b0-49ef-be29-14a98361a3b0.png">

mirisuzanne commented:
I like these ideas. I'd probably have time to implement some of it later this week unless @dvdherron or @stacyk would like to take it on?

stacyk commented:
@SondraE @mirisuzanne I like your proposal. Looking at our projects, I wonder if a potential client would want to see more sectors or more projects within a wider sector category. For instance: 

If we have `Health` we would tag MedCurbside, CoachHub, QuarqRace, QuarqNet

But if the sector was `Medical`, then we'd probably only have MedCurbside and maybe CoachHub?

Then `Fitness` could be QuarqNet, QuarqRace, CoachHub

or `Sports` could be Top Dog and QuarqNet/Race

I think the "Work Sample" tag is fine because you can get to these from the blog, and that tag will link you over to the other case studies. but I wonder if we would want to try to link them to the Work page instead? I know this is different from how the other tags work though.

dvdherron commented:
I like your proposal too Sondra. @mirisuzanne I probably wouldn't be able to get started on this until next week?

SondraE commented:
@stacyk Great questions/thoughts about what the sector tags should be. I've listed ideas below. We can certainly change these and add more too.


Design
- Adobe Game

Technology or maybe Big Tech
- Adobe Game
- Shopify Training
- Mozilla Videos
- MozTrap
- MetaDeploy
- Google (will we add this?)

e-Commerce
- Shopify Training

Analytics and/or Data
- Workflow Builder
- Expression Builder
- Second Measure

Finance
- CFOshare

Education
- TimeDesigner
- Emmy's Workshop

Health or Wellness
- MedCurbside
- QuarqRace
- QuarqNet
- CoachHub

Sports
- Top Dog
- QuarqRace
- QuarqNet

Manufacturing
- GSM

Consumer Web App (kind of a sector?)
- Top Dog
- Second Measure (?)
- MedCurbside
- Quarq Race & Net
- CoachHub

dvdherron commented:
I started to implement some of these layout changes in the summaries @SondraE has started working on. I didn't get to Shopify. 

- Added placeholder dates to the front matter. We would have to double check what the exact start dates are if that's how we we want to display them. Otherwise, I think the date macro would have to be edited to only show Month/Year combinations?
- Added a `sectors` macro to include sectors in the sub-header
- The images in the templates I edited are all using the same placeholder image for now. 

Do we want to differentiate the sectors tags somehow @stacyk @SondraE ? @mirisuzanne  mentioned maybe adding some prefix (client?) to the new set of tags.

I think on some of these the column size might need to be adjusted but it's a good start for implementing the mockups above.

mirisuzanne commented on content/work/cfoshare.md:
```diff
@@ -20,18 +21,68 @@ press:
     venue: *client
 links:
   site: https://cfoshare.org/
+sector:
+  - Finance
+  - Accounting
+tags:
+  - Finance
+  - Accounting
```
If we end up using the same names, and we prefix the tags - we could probably auto-generate sector names from tag names… (I guess we'll see where all that lands)

SondraE replied:
@mirisuzanne @dvdherron @stacyk 

- If we could change the date macro to allow display of only the years for client projects, I think that would be ideal. Any other date info seems not as useful to visitors, and looking back through all the projects to find accurate start/end months would take some doing. (I'm willing to hear arguments to the contrary, though.)
- It would be nice if we could just put in tags and auto-generate sector names. Or maybe we remove the sector information from the sub-header and have the tags say "Accounting Sector" and Financial Sector."

SondraE commented:
@davisagli @stacyk

Based on Stacy's suggestion this morning, I think we should do a few more things on this PR and then push what we have because rewriting all the client summaries could take me quite awhile. For example, for client project I didn't work on, I will need to gather the updated summary info from other Birds.

Here's my suggestion for what to finish on this PR.

- Finalize the summaries and images for shopify/, cfoshare/, workflow-builder/, expression-builder/
- Decide on sector and/or tag format, and add those to all work projects
- Update date macro to display just the year on client projects, and add years to all work projects

Does that seem like a good plan?

dvdherron commented:
@SondraE Your plan sounds like a good plan to implement this on a few of the summaries and wrap up this PR. 

- For the sector/tag issue, I think I lean more on the side of not including "sector" in the tag name. I'm wondering if it would make a difference to a potential client if a tag said "Finance" versus "Finance Sector." Like @stacyk said, we can try either way and change later. 
- I think only displaying the years on the client projects makes sense. The full, specific dates probably won't matter so much to visitors. I don't think this will require too much wrangling of the date macro logic but I will have to look again. 

> @dvdherron @stacyk
> 
> Based on Stacy's suggestion this morning, I think we should do a few more things on this PR and then push what we have because rewriting all the client summaries could take me quite awhile. For example, for client project I didn't work on, I will need to gather the updated summary info from other Birds.
> 
> Here's my suggestion for what to finish on this PR.
> 
>     * Finalize the summaries and images for shopify/, cfoshare/, workflow-builder/, expression-builder/
> 
>     * Decide on sector and/or tag format, and add those to all work projects
> 
>     * Update date macro to display just the year on client projects, and add years to all work projects
> 
> 
> Does that seem like a good plan?

SondraE commented:
@stacyk I'm not sure whether workflow builder had a mobile screen size view. Do you know? Also do you happen to know if we have any additional screenshots of workflow builder other than the one that is the header on the project page, or whether we can get any more at this point?

dvdherron commented:
I looked into some of the changes that might have to be made to tweak the date format for these summaries. 

For projects that start and end within one year, would we rather have the meta data read "for Client X in 2020" or "for Client X from Mar-Oct 2020?"

I agree that the full date ( e.g. October 23, 2020 )  is too much info, but maybe leaving the month in is okay? 

I think if we choose to only display the years for these summaries, we would need to change the prefix logic ( in `utility.macros` ). 

And I would _think_ changing the date format would happen here in `meta.njk` but I've so far haven't seen anything change even if I change format to just 'year'. Am I making the change in the wrong place @stacyk ?

```
>  {% if author or client %}
>         ...
>         {{ post.pubdate(
>           start=page.date,
>           end=end,
>           format='range' if oss else 'long'
>         ) }}
>        ...
>       </p>
```

SondraE commented:
@dvdherron @stacyk 
Thanks for looking into what it would take to modify the date format! 
For projects that start and end in the same year, I think it could be useful to show the months.
For projects that span multiple years, we could leave in the month, but if we do I'd like to make sure the months are accurate, and I think that information would take some time and research to find for every project. For the sake of time (and maybe skimmability) I'd lean toward not showing the months for the multi-year projects, but I'm willing to be out voted on that.

stacyk commented:
> And I would _think_ changing the date format would happen here in `meta.njk` but I've so far haven't seen anything change even if I change format to just 'year'. Am I making the change in the wrong place @stacyk ?

@dvdherron I think we want to look at the byline macro in post.macros.njk  ... If you look at the work archive, it displays as we want (I think, if I followed htis correctly)

stacyk commented:
![Screen Shot 2021-08-18 at 10 22 00 AM](https://user-images.githubusercontent.com/1581694/129927353-bc917f66-4366-489d-9b47-0af9f51dcd42.jpg)

Just spoke with Sondra, and she confirmed we don't need to show the sector name in the subtitle on the case study single post page, but we can keep it on the archive page since we aren't showing tags there (or will we in the future?)

dvdherron commented:
@SondraE @stacyk 
- I changed the layout on the workflow-builder and expression-builder pages.
- Also removed the logic and yml data for sectors

The last bit remaining is getting the years to display how we want. I didn't make it to that today but I can pick back on Monday. 


![image](https://user-images.githubusercontent.com/41588129/130263896-d6f78ca2-97e0-4c12-9ee6-873dec9db7fe.png)
===
![image](https://user-images.githubusercontent.com/41588129/130264003-604a3629-607a-41b7-b10d-cd9f9ebd8d31.png)

SondraE commented:
@dvdherron Looking so good! I'm excited about having this content on the Work pages!

A few notes. 

1. I noticed the line length of the second column looks shorter on CFOshare than on Shopify. This isn't a big deal, but since some of the CFOshare word breaks look odd to me, I thought I'd ask if you know why.
<img width="1440" alt="Screen Shot 2021-08-25 at 10 26 34 AM" src="https://user-images.githubusercontent.com/8933115/130830022-194579b9-8e18-415b-ba8f-61cce52b4b55.png">
<img width="1440" alt="Screen Shot 2021-08-25 at 10 29 07 AM" src="https://user-images.githubusercontent.com/8933115/130830030-1e51195f-275d-4196-a07f-f441731e6b25.png">


2. Some of the dates are ordered backwards. They look like this: 2016 - 2014 instead of like this: 2014 - 2016. That could definitely be my own data entry error. :P


3. Now that we are displaying the years worked, these clients appear to be listed in a random order. Probably makes sense to order them with the most recent projects at the top. Ordering by date is tricky, though. If a project like TimeDesigner has "2016 - 2018" and then we want to add "& 2021 - 2022" does it move to the top of the list? Also tricky to know how to order if projects that have the same dates. Also do we split up projects for the same company if they are separated in time by several years? (Right now, projects are grouped together under the company header.) We don't add projects often. Maybe we use a handpicked order instead of something automatic? Happy to hear others' ordering ideas. @mirisuzanne @stacyk @jgerigmeyer 


4. @mirisuzanne @stacyk @jgerigmeyer As you all review, I would appreciate edits and additions to the bulleted lists of what we did for the four projects we're updating in this PR.


5. One more thought. I wonder whether we should show the quotes before the What We Did section. It is so easy to miss them right now and, according to user interviews I've run and research I've read, visitors may find those quotes a little more important/convincing than the list of services OddBird provided.

stacyk commented on content/work/workflow-builder.md:
```diff
@@ -37,15 +38,58 @@ press:
+{{ layout.title('What We Did') }}
+
+
+{% call embed.media_block(
+  media=embed.img(
+  src='pages/workflow-sample.jpg',
+   alt='map of all workflow block colors',
+   attrs={
+    'width':'1280',
+    'height':'730'
+   }),
+   name='desktop-work'
+) %}
```
@jgerigmeyer would you mind taking a screenshot or two of Workflow Builder at at least 1600px wide and maybe around 100px high? Something cool happening on the screen :) Can be similar to what we show in the hero, but maybe with the toolbar showing on a selected card or something.

jgerigmeyer replied:
@stacyk One of these (I didn't crop yet)? I have it running now, so if you want something different let's jump on a call.

![Screen Shot 2021-08-31 at 3 07 11 PM (3)](https://user-images.githubusercontent.com/552316/131561932-d825f355-f6f2-4753-8911-117366274f43.png)
![Screen Shot 2021-08-31 at 3 07 15 PM (3)](https://user-images.githubusercontent.com/552316/131561936-7b5bbda7-51dc-4996-971e-11b15df12b77.png)
![Screen Shot 2021-08-31 at 3 07 51 PM (3)](https://user-images.githubusercontent.com/552316/131561941-33603d81-5e78-4858-88ab-c69b4c8b6019.png)
![Screen Shot 2021-08-31 at 3 10 05 PM (3)](https://user-images.githubusercontent.com/552316/131561942-d9f6f74c-eb4b-45bf-8d7b-9f3bacb8845b.png)

stacyk replied:
> @stacyk One of these (I didn't crop yet)? I have it running now, so if you want something different let's jump on a call.

One of these will work. I am going to photoshop out the top and right side. Thanks for sending!

jgerigmeyer commented on content/work/cfoshare.md:
```diff
@@ -20,18 +21,65 @@ press:
+    notch='0',
+    content=embed.img(
+      src='pages/cfo-phone.jpg',
+      alt='trig race screen',
```
🤔

jgerigmeyer commented on content/work/expression-builder.md:
```diff
@@ -47,6 +52,48 @@ summary: |
+{% call embed.media_block(
+  media=embed.img(
+      src='pages/expression-sample.jpg',
+      alt='trig race screen',
```
🤔

jgerigmeyer commented on content/work/shopify.md:
```diff
@@ -29,5 +33,47 @@ summary: |
+    notch='0',
+    content=embed.img(
+      src='pages/shopify-phone.jpg',
+      alt='trig race screen',
```
🤔

jgerigmeyer commented on content/work/shopify.md:
```diff
@@ -29,5 +33,47 @@ summary: |
+  media=embed.screen(
+    notch='0',
+    content=embed.img(
+      src='pages/shopify-phone.jpg',
```
Is this a helpful image? It doesn't make sense to me -- why is Miriam training them on a mobile phone? 😄

dvdherron replied:
Sondra swapped images here:
![bild](https://user-images.githubusercontent.com/41588129/132854541-4340fa69-0942-457e-9182-326ac38e8299.png)

jgerigmeyer commented on content/work/workflow-builder.md:
```diff
@@ -37,15 +38,58 @@ press:
+
+{% call embed.media_block(
+  media=embed.img(
+  src='pages/workflow-sample.jpg',
```
Needs updated image.

jgerigmeyer commented on content/_includes/utility.macros.njk:
```diff
@@ -232,9 +232,9 @@ params:
+      {{ start | getDate(ranges[same].start or 'year') }}
       {{- '--' | typogr | safe -}}
-      {{ end | getDate(ranges[same].end or 'month_year') }}
+      {{ end | getDate(ranges[same].end or 'year') }}
```
I'm curious what the goal of this change was -- it has an impact throughout the site, because this macro is used so broadly. (Not saying it's a bad change throughout; but noticing that it does touch a lot of pages.)

SondraE replied:
@jgerigmeyer I asked for that change because I didn't think the month start and end would be that useful to visitors and because I didn't want to have to spend the time figuring out which months we started and ended each of our projects. If it is causing problems elsewhere, though, I could be convinced to keep month_year.

jgerigmeyer replied:
Sure, that makes sense. I'm not sure it's causing problems elsewhere, I just noticed that the change is global throughout many dates on the site.

jgerigmeyer reviewed (commented):
Looks like an improvement! I think this isn't quite done though?

jgerigmeyer commented on content/work/shopify.md:
```diff
@@ -37,19 +37,18 @@ summary: |
+    src='work/shopify/shopify.jpg',
+    alt='Miriam speaking in the UX Summer Shopify Series',
+    attrs={
+      'width':'1800',
+      'height':'807'
```
I didn't think `width` and `height` were needed anymore. @mirisuzanne?

mirisuzanne replied:
I think you're right, the new responsive-image tool doesn't require these?

dvdherron replied:
removed these.

stacyk commented on content/work/expression-builder.md:
```diff
@@ -3,9 +3,10 @@ title: Expression Builder
+end: 2019-05-14
 image:
-  src: projects/aunalytics/exbldr.jpg
+  src: work/expression-builder/exbldr.jpg
```
@SondraE I noticed we have a few images like this (I think exbldr2.jpg and 3.jpg) that I didn't see us using. Can you confirm, and if we aren't we can probably delete those from the project.

stacyk replied:
Also, along these lines, I think we aren't using any of these:

src/images/work/quarqnet/mockup2.png
src/images/work/quarqnet/implementation1.jpg
src/images/work/quarqnet/implementation2.jpg
src/images/work/quarqnet/qasper-splash.jpg
src/images/work/quarqnet/quarqnet-sketch-mobile.jpg
src/images/work/quarqnet/tabs.jpg
src/images/work/quarqrace/trig-mobile.jpg
src/images/work/quarqrace/trig-splash.jpg


I didn't check all of the images, and I assume we aren't using some coachhub and maybe adobe images.

jgerigmeyer reviewed (approved):
This looks good to me! I'd like to test out the new diffing of built HTML on this before merge.

jgerigmeyer commented:
@jerivas It works! (Though it took me a bit to figure out that I needed the `..` compare instead of the usual `...`): https://github.com/oddbird/oddleventy-built/compare/main..work-summaries

mirisuzanne commented on src/scss/patterns/_media.scss:
```diff
@@ -102,3 +102,12 @@
   @include tools.square(100%);
 }
+
+[data-media='desktop-work'] {
```
I can't find where this is being used?

stacyk replied:
I thought it was on examples where we aren't showing a mobile screen, like on Workflow Builder Case Study in the what we do section?

stacyk replied:
![Screen Shot 2021-09-13 at 11 27 58 AM](https://user-images.githubusercontent.com/1581694/133121710-29b89245-d3f9-401c-8b8d-d9f5813f43a4.jpg)

dvdherron replied:
@mirisuzanne @stacyk yeah, it's being used on the expression-builder, shopify and workflow-builder pages that don't have a mobile-specific sample.

mirisuzanne replied:
aha, ok, looks great!

mirisuzanne reviewed (approved):
It all looks good to me! But I wasn't sure what the 'desktop-work' styles were applied to…
